core/src/packet.rs

```diff
@@ -1,8 +1,8 @@
-use std::{mem, num::NonZeroU8};
+use std::{ffi::c_void, mem, num::NonZeroU8, ptr};
 
 use crate::error::{SendError, SendErrorKind};
 
-pub use sys::generated_safe::OperationKind;
+pub use sys::{generated_safe::OperationKind, tb_packet_t as Raw};
```

Comment:
Added it to follow the current `master`:

<img width="659" alt="Снимок экрана 2024-11-11 в 05 06 24" src="https://github.com/user-attachments/assets/e87743c3-140e-4e55-8047-29b7689574c6">

Should we remove all of these `Raw` re-exports?

Reply:
I think we can use `pub(crate)` and use these re-exports in places like [1](https://github.com/tigerbeetle-rust/tigerbeetle-unofficial/blob/c644b73969f2e927b03462a1d4964201a411726b/core/src/callback.rs#L92), [2](https://github.com/tigerbeetle-rust/tigerbeetle-unofficial/blob/c644b73969f2e927b03462a1d4964201a411726b/core/src/lib.rs#L26)

Reply:
No, there's no reason to make this change. The difference between those structs is that `tb_account_t` and `tb_transfer_t` are intended to be used as [PODs](https://en.wikipedia.org/wiki/Passive_data_structure) from the tigerbeetle core while `tb_packet_t` contains pointers, which `Packet` may interpret as valid and exposing the `Raw` structure might mess up with this type invariant, which is why `Packet::raw(_mut)` methods are private. Even the name resolution of this struct becomes `tigerbeetle_unofficial_core::Raw` if you check, because such type exports weren't intended.